### PR TITLE
feat(v2): publish snapshots

### PR DESCRIPTION
**Pull request by scottgerring**
**Issue #, if available:**

## Description of changes:

**Checklist**

 

* [ ] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda-java/#tenets)
* [ ] Update tests
* [ ] Update docs
* [ ] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/publish-v2-snapshot.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Publish v2 snapshots to the Maven Central repository. Runs on merge to v2.
+on:
+  pull_request:
+    branches:
+      - v2
+  push:
+    branches:
+      - v2
+      -
```
Comment (jeromevdl) [inline .github/workflows/publish-v2-snapshot.yml:9]:
```suggestion
```

File: .github/workflows/publish-v2-snapshot.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name: Publish v2 snapshots to the Maven Central repository. Runs on merge to v2.
+on:
+  pull_request:
+    branches:
+      - v2
```
Comment (jeromevdl) [inline .github/workflows/publish-v2-snapshot.yml:5]: I don't think we want this, on each PR, we just want the merged PR

```suggestion
```

Comment (scottgerring) [reply .github/workflows/publish-v2-snapshot.yml:5]: Just there to test for now!

File: spotbugs-exclude.xml [hunk 3]
```diff
@@ -8,6 +8,22 @@
        https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html
   -->
 <FindBugsFilter>
+    <!-- These open matches are new rules that we have not yet addressed. Let's block them generically, and come
```
Comment (scottgerring) [inline spotbugs-exclude.xml:11]: @jeromevdl i'm not sure why spotbugs suddenly has more rules, but it does. As the purpose of this is to just get V2 snapshots + docs publishing i've excluded them generically (e.g. not bound to a particular class or field) for now.

File: .github/workflows/docs-v2-snapshot.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+name: Docs
+on:
+  pull_request:
+    branches:
+      - v2
```
Comment (jeromevdl) [inline .github/workflows/docs-v2-snapshot.yml:5]:
```suggestion
```

not on PRs

File: .github/workflows/publish-v2-snapshot.yml (hunk 2 above)
Comment (jeromevdl) [reply .github/workflows/publish-v2-snapshot.yml:5]: so you can remove it now ;)

File: spotbugs-exclude.xml (hunk 3 above)
Comment (jeromevdl) [reply spotbugs-exclude.xml:11]: ok

**Review by jeromevdl — CHANGES_REQUESTED** [review]
just remove the triggers on PRs and we're good

**Review by jeromevdl — APPROVED** [review]
LGTM

**Pull request merged**